### DISCUSSION TITLE
Any idea for implementing Material 3 carousel with swiper?

### DISCUSSION THREAD

**Discussion by shayandrg** [nolimits4web/swiper#7185] [Q&A]
### Check that this is really a bug

- [x] I confirm

### Reproduction link

https://swiperjs.com/demos#effect-creative

### Bug description

hey guys

i want to build material 3 carousel with swiper and I have tried many techniques but failed.

https://m3.material.io/components/carousel/guidelines

i tried to change slide width width dom modification on **setTranslate** width swiper.update() and it changes the slides width but swiper getting bug for calculating slides move and many other issues

any ideas?

### Swiper version

8

### Platform/Target and Browser Versions

ubuntu 22

### Validations

- [X] Follow our [Code of Conduct](https://github.com/nolimits4web/swiper/blob/master/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://swiperjs.com/swiper-api).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
- [X] Make sure this is a Swiper issue and not a framework-specific issue

### Would you like to open a PR for this bug?

- [ ] I'm willing to open a PR

**Comment by nolimits4web** [comment]
This looks cool, I'll try to work on it, but if I succeed that will definitely go to UI Initiative as a premium plugin 😊
